service/src/test/java/org/sunbird/learner/actors/bulkupload/TextbookTocActorTest.java

```diff
@@ -24,145 +38,240 @@
+    InputStream stream = new ByteArrayInputStream(data.getBytes());
+    byte[] byteArray = IOUtils.toByteArray(stream);
+    request.getRequest().put(JsonKey.DATA, byteArray);
+    request.put(JsonKey.DATA, byteArray);
     request.setOperation(TextbookActorOperation.TEXTBOOK_TOC_UPLOAD.getValue());
-
     toc.tell(request, probe.getRef());
-    Response res = probe.expectMsgClass(duration("100 second"), Response.class);
-    Assert.assertNotNull(res);
-    Assert.assertEquals("OK", res.getResponseCode().name());
+    if (error) {
+      ProjectCommonException res =
+          probe.expectMsgClass(duration("10 second"), ProjectCommonException.class);
+      return res;
+    }
+    Response response = probe.expectMsgClass(duration("10 second"), Response.class);
+    return response;
+  }
+
+  private void mockRequiredMethods(boolean error) {
+    when(TextBookTocUtil.getRelatedFrameworkById(Mockito.anyString())).thenReturn(new Response());
+    when(TextBookTocUtil.readHierarchy(Mockito.anyString())).thenReturn(getReadHierarchy(error));
+    when(TextBookTocUtil.readContent(Mockito.anyString())).thenReturn(getReadContentTextbookData());
   }
 
-  private static Response getTbWithoutChildren() throws IOException {
-    return mapper.readValue(TB_WITHOUT_CHILDREN, Response.class);
+  private Response getReadHierarchy(boolean error) {
+    Response response = new Response();
+    List<String> tocData = new ArrayList<>();
+    Map<String, Object> content = new HashMap<>();
+    if (error) {
+      content.put(JsonKey.IDENTIFIER, "do_112678881305763840115");
+    } else {
+      content.put(JsonKey.IDENTIFIER, "do_1126788813057638401122");
+    }
+    tocData.add("2019");
+    content.put(JsonKey.DIAL_CODES, tocData);
+    content.put(JsonKey.CHILDREN, new ArrayList<>());
+    response.put(JsonKey.CONTENT, content);
+    return response;
   }
 
-  private static Response getSuccess() {
+  private Response getReadContentTextbookData() {
     Response response = new Response();
-    response.put("content_id", "do_11265332762881228812868");
+    Map<String, Object> textBookdata = new HashMap<>();
+    Map<String, Integer> reserveDialCodes = new HashMap<>();
+    reserveDialCodes.put("2019", 1);
+    textBookdata.put(JsonKey.RESERVED_DIAL_CODES, reserveDialCodes);
+    textBookdata.put(JsonKey.CONTENT_TYPE, CONTENT_TYPE);
+    textBookdata.put(JsonKey.MIME_TYPE, "application/vnd.ekstep.content-collection");
+    textBookdata.put(JsonKey.NAME, "test");
+    response.put(JsonKey.CONTENT, textBookdata);
     return response;
   }
+
+  private static String getFileAsString(String fileName) {
+    File file = null;
+
+    try {
+      file = new File(TextbookTocActorTest.class.getClassLoader().getResource(fileName).getFile());
+      Path path = Paths.get(file.getPath());
+      List<String> fileData = Files.readAllLines(path);
+      StringBuffer result = new StringBuffer();
+      for (String line : fileData) {
+        result.append(line);
```

Comment:
does this maintain new-line? probably you need to append new-line as well.

Reply:
Done